src/libraries/System.Linq.Parallel/src/System/Linq/Parallel/Scheduling/QueryTaskGroupState.cs

```diff
@@ -127,7 +127,7 @@ internal void QueryEnd(bool userInitiatedDispose)
 
                     // if all the exceptions were OCE(externalToken), then we will propagate only a single OCE(externalToken) below
                     // otherwise, we flatten the aggregate (because the WaitAll above already aggregated) and rethrow.
-                    if (!allOCEsOnTrackedExternalCancellationToken)
+                    if (!allOCEsOnTrackedExternalCancellationToken || flattenedAE.InnerExceptions.Count == 0)
```

Comment:
If I changed the repro to throw an `AggregateException` populated with inner exceptions would the bug still reproduce?

Reply:
> If I changed the repro to throw an AggregateException populated with inner exceptions would the bug still reproduce?

No (unless they were all themselves empty AggregateExceptions)